Readme.md

```diff
@@ -91,22 +91,44 @@ This plugin can integrate with `fastify-auth` by following this example:
 
 ```js
 const fastify = require('fastify')()
+const auth = require('fastify-auth')
 const bearerAuthPlugin = require('fastify-bearer-auth')
 const keys = new Set(['a-super-secret-key', 'another-super-secret-key'])
 
-fastify.register(bearerAuthPlugin, { addHook: false, keys})
-
-fastify.route({
-  method: 'GET',
-  url: '/multiauth',
-  preHandler: fastify.auth([
-    fastify.allowAnonymous,
-    fastify.verifyBearerAuth
-  ]),
-  handler: function (_, reply) {
-    reply.send({ hello: 'world' })
-  }
-})
+const server = async function () {
+
+  await fastify
+    .register(auth)
+    .register(bearerAuthPlugin, { addHook: false, keys})
+    .decorate('allowAnonymous', function (req, reply, done) {
+      if (req.headers.authorization) {
+        return done(Error('not anonymous'))
+      }
+      return done()
+    })
+
+  fastify.route({
+    method: 'GET',
+    url: '/multiauth',
+    preHandler: fastify.auth([
+      fastify.allowAnonymous,
+      fastify.verifyBearerAuth
+    ]),
+    handler: function (_, reply) {
+      reply.send({ hello: 'world' })
+    }
+  })
+
+  fastify.listen({port: 8000}, (err) => {
```

Comment:
Could we use await here to avoid mixing too many patterns?

Reply:
Updated in https://github.com/fastify/fastify-bearer-auth/pull/102/commits/3845b7e3e3f52f0ce8291f0817766d8c8e5bb3d7

Reply:
If there is a callback, a promise is not returned 😞